### PR TITLE
core: remove deprecated gulp-util dependency

### PR DESCRIPTION
**Pull request by mjeanroy**
Hi,

Since `gulp-util` has been deprecated, this PR replace `gulp-util` with alternative individual modules. Please see: https://github.com/gulpjs/gulp-util

### CODE REVIEW

File: example/fail.js [hunk 1]
```diff
@@ -3,7 +3,7 @@
 // npm install gulp gulp-eslint
 
 const gulp = require('gulp');
-const gulpUtil = require('gulp-util');
+const log = require('fancy-log');
```
Comment (shinnn) [inline example/fail.js]: Can you rename this to `fancyLog`? I prefer explicitness.

Comment (mjeanroy) [reply example/fail.js]: Done ;)

**Comment by demurgos** [comment]
@shinnn
Your request was addressed and the rest of the PR looks good. Could you merge this PR and publish an update to _npm_? I am currently going over multiple gulp plugins to fix the same issue. Some of them have dependencies on _gulp-eslint_ so it would be nice to have it solved quickly for this library so the other plugins can be updated. Thanks :)

File: test/util.js [hunk 2]
```diff
@@ -216,7 +216,7 @@ describe('utility methods', () => {
 		it('should default to gutil.log', () => {
```
Comment (ninbryan) [inline test/util.js]: shouldn't the test message be `it('should default to fancyLog, () => {`

Comment (ninbryan) [reply test/util.js]: the jsDoc should also be updated where `[writable=gulp-util.log]` is specified
- https://github.com/mjeanroy/gulp-eslint/blob/refactor-remove-deprecated-gulp-util-dependency/index.js#L167
- https://github.com/mjeanroy/gulp-eslint/blob/refactor-remove-deprecated-gulp-util-dependency/index.js#L181

Comment (shinnn) [reply test/util.js]: Yes, what @ninbryan said. Good catch.

Comment (mjeanroy) [reply test/util.js]: @ninbryan Good catch, thanks!

**Comment by mjeanroy** [comment]
@shinnn I also updated the `package-lock.json` (I use node 9.3.0 / npm 5.5.1).

**Comment by demurgos** [comment]
Regarding the lock files, I also used always used them but I had a discussion recently with some other maintainers. They are extremely valuable to get reproducible builds for applications, but it may be better to avoid them for libraries. Since they lock dependencies for developers and CI but not consumers, it means that error with transitive dependencies will always affect consumers first instead of being caught by CI.

**Comment by mjeanroy** [comment]
@shinnn I updated `CHANGELOG.md`. Regarding `package-lock.json`, do you want me to revert this change?

**Comment by shinnn** [comment]
> Regarding `package-lock.json`, do you want me to revert this change?

No. Just ignore @demurgos's comment, I've already deleted it though.

**Comment by demurgos** [comment]
Sorry for my comment about the lock file update. The use or not of lock files has no clear answer and it was not the place for this PR to discuss it.

@mjeanroy 
It's good that you updated the lock file.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## 4.0.2
+
+* Remove deprecated [`gulp-util`](https://github.com/gulpjs/gulp-util) dependency and use individual modules instead.
```
Comment (shinnn) [inline CHANGELOG.md]: Remove the last `.` for consistency.

Comment (mjeanroy) [reply CHANGELOG.md]: Done.

**Pull request merged**